app/models/author.rb

```diff
@@ -1,5 +1,17 @@
+# frozen_string_literal: true
+
 class Author < ApplicationRecord
-  def discount() 10 end
+  # This translates params from GitHub API
+  alias_attribute :title, :name
+  alias_attribute :body, :biography
```

Comment:
Interesting, thanks. I have sort of worked with the idea that the model handles data going to the database. Not to disagree with you, but just to present why I did that.
To bring this transformation back to the controller I could make a before_action:
```suggestion
  def transform_params
    params[:author] = {
      name: params[:issue][:title],
      biography: params[:issue][:body]
    }
  end
```

Reply:
Ya, totally. There are a few different approaches to where this sort of data should live. 

At Smile, we keep this sort of transformation out of the model as it's common for the same model to be used in a wide variety of situations (different API endpoints, rake tasks, background processing jobs, etc) and the input data formats in all those cases can be very different. 

Rather than have a bunch of different transform methods on the model, we find the code is easier to maintain if we extract that logic to the calling code (e.g. the controller in this case) as it moves the transformation "closer" to the specific case where a certain set of input data will be received.